subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -635,8 +635,62 @@ int ull_adv_sync_init(void)
 
 int ull_adv_sync_reset(void)
 {
+	struct lll_adv_sync *lll_sync;
+	struct ll_adv_sync_set *sync;
+	struct ll_adv_set *adv;
+	uint8_t handle;
 	int err;
 
+	for (handle = 0U; handle < BT_CTLR_ADV_SET; handle++) {
+		adv = ull_adv_is_created_get(handle);
+		if (!adv) {
+			continue;
+		}
+
+		lll_sync = adv->lll.sync;
+		if (!lll_sync) {
+			continue;
+		}
+
+		sync = HDR_LLL2ULL(lll_sync);
+
+		if (!sync->is_started) {
+			sync->is_enabled = 0U;
+
+			continue;
+		}
+
+		err = sync_remove(sync, adv, 0U);
+		if (err) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+int ull_adv_sync_reset_finalize(void)
+{
+	struct lll_adv_sync *lll_sync;
+	struct ll_adv_set *adv;
+	uint8_t handle;
+	int err;
+
+	for (handle = 0U; handle < BT_CTLR_ADV_SET; handle++) {
+		adv = ull_adv_is_created_get(handle);
+		if (!adv) {
+			continue;
+		}
+
+		lll_sync = adv->lll.sync;
+		if (!lll_sync) {
+			continue;
+		}
+
+		lll_adv_data_reset(&lll_sync->data);
+		adv->lll.sync = NULL;
+	}
```

Comment:
```suggestion
```

Reply:
Referring to comment: https://github.com/zephyrproject-rtos/zephyr/pull/37333#discussion_r680801307

I suggest to do not remove the function. Just change it to set all adv->lll.sync to NULL.
So this part may be removed:
```
lll_sync = adv->lll.sync;
if (!lll_sync) {
	continue;
}
lll_adv_data_reset(&lll_sync->data);
```